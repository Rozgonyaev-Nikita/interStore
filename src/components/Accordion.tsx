import React from "react";

interface AccordionProps {
  title: string;
  children: React.ReactNode;
  data?: IData[];
}
interface IData {
  title: string;
}
const Accordion = (props: AccordionProps) => {
  const { title, children, data = [] } = props;

  return (
    <div className="accordion" data-testid="div-accord">
      <h3 className="accordion-title">{title}</h3>
      <div className="accordion-content">{children}</div>
      {data.length ? (
        data.map((dat, index) => <p key={index}>{dat.title}</p>)
      ) : (
        <p>List void</p>
      )}
      <title>Testing</title>
    </div>
  );
};

export default Accordion;
