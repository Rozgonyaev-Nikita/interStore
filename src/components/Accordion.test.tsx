import { describe, expect, expectTypeOf, test } from "vitest";
import { logRoles, render, screen } from "@testing-library/react";
import Accordion from "./Accordion";

const data = [
  { title: "1" },
  { title: "2" },
  { title: "3" },
  { title: "4" },
  { title: "5" },
];

interface Iput {
  input: number;
  output: number;
}

describe("Accordion test", () => {
  test("should show title all the time", () => {
    render(
      <Accordion title="Testing" data={data}>
        <div>
          <h4>Content</h4>
          <div role="generic">fh</div>
        </div>
      </Accordion>
    );

    // expect(screen.getByText(/Testing/i)).toBeDefined();
    // expect(screen.getByText(/Content/i)).toBeDefined();
    expect(screen.getByText(/ent/));
  });
  test("test role data", () => {
    render(
      <Accordion title="Testing">
        <div>karp</div>
      </Accordion>
    );

    expect(screen.queryByText(/List void/i)).not.toBeNull();
  });
  test("test button role", () => {
    render(
      <Accordion title="Testing">
        <button>cazan</button>
      </Accordion>
    );

    expect(screen.getByRole("button", { name: "cazan" }));
  });
  test("test title", () => {
    render(
      <Accordion title="Testing">
        <button title="buto">cazan</button>
      </Accordion>
    );

    // expect(screen.queryByTitle("Testing"));
    const titl = screen.queryByTitle("buto");
    expect(titl).not.toBeNull();
  });

  test("karp", () => {
    expect(2 + 2).toBe(4);
  });
  describe.each([
    { input: 1, output: 2 },
    { input: 4, output: 5 },
    { input: 3, output: 4 },
    { input: 1125, output: 1126 },
  ])("Accordion test", ({ input, output }: Iput) => {
    test("kfg", () => {
      expect(input + 1).toBe(output);
    });
  });
  test("should show title all the timejkS", () => {
    render(
      <Accordion title="Testing">
        <div title="di">
          <h3>Content</h3>
          <div role="generic">fh</div>
        </div>
      </Accordion>
    );

    // expect(screen.getByText(/Testing/i)).toBeDefined();
    // expect(screen.getByText(/Content/i)).toBeDefined();
    const scre = screen.getByTestId("div-accord");
    expect(scre).toMatchSnapshot();
  });
});
