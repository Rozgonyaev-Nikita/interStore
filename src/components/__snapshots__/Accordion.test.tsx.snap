// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Accordion test > should show title all the timejkS 1`] = `
<div
  class="accordion"
  data-testid="div-accord"
>
  <h3
    class="accordion-title"
  >
    Testing
  </h3>
  <div
    class="accordion-content"
  >
    <div
      title="di"
    >
      <h3>
        Content
      </h3>
      <div
        role="generic"
      >
        fh
      </div>
    </div>
  </div>
  <p>
    List void
  </p>
  <title>
    Testing
  </title>
</div>
`;
